OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs

```diff
@@ -183,7 +186,29 @@ Order IIssueDeployOrder.IssueDeployOrder(Actor self, bool queued)
 			return new Order("GrantConditionOnDeploy", self, queued);
 		}
 
-		bool IIssueDeployOrder.CanIssueDeployOrder(Actor self, bool queued) { return !IsTraitPaused && !IsTraitDisabled; }
+		bool IIssueDeployOrder.CanIssueDeployOrder(Actor self, bool queued)
+		{
+			if (IsTraitPaused || IsTraitDisabled || self.IsDead || self.Disposed)
+				return false;
+
+			if (queued || !Info.SmartDeploy || DeployState == DeployState.Undeployed || DeployState == DeployState.Undeploying)
+				return true;
+
+			foreach (var actor in self.World.Selection.Actors)
```

Comment:
I would have expected some sort of logic where all units would take on the deploy state of one main lead unit. But I don't see that directly in this loop.

Couldn't this cause toggles both ways for different units in the same selection?

There is only one type of deploy? I.e. if you have both a chinook and apc selected - would both undeploy/deploy? Or that is prob not deploy but carry.

Reply:
a chinook doesn't use `GrantConditionOnDeploy`. What is a lead unit?